File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        i = 0
        j = 0
        output = 0
        zeroes = 0

        while (j < len(nums)):
            if (nums[j] == 0):
                zeroes += 1

            while (zeroes > k):
                if (nums[i] == 0):
                    zeroes -= 1
                i += 1

            output = max(output, j - i + 1)
            j += 1

        return output


        

File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.js
var longestOnes = function(nums, k) {
    let output = 0
    let i = 0
    let j = 0
    let flippedZeroes = 0

    while (j < nums.length) {

        if (nums[j] == 0) flippedZeroes += 1

        while (flippedZeroes > k) {
            if (nums[i] != 1) flippedZeroes -=1
            i += 1
        }

        output = Math.max(output, j - i + 1)
        j += 1
    }

    return output
};